contracts/crowdloan/Crowdloan.sol

```diff
@@ -142,31 +169,13 @@ contract Crowdloan is Initializable, ICrowdloan, ReentrancyGuard {
         );
 
         if (termsContract.getLoanStatus() < TermsContractLib.LoanStatus.REPAYMENT_CYCLE) {
-            termsContract.startLoan();
+            termsContract.startRepaymentCycle(totalCrowdfunded); // TODO(Dan): change this to be the amount actually raised
```

Comment:
this should be
`termsContract.startRepaymentCycle(_getPrincipalToken().balanceOf(address(this)))`,
which simply takes the balance at this point in time of the crowdloan contract.
This also removes the need to track `totalCrowdfunded` within the fund and refund functions.

Reply:
@adibas03 Ah yes I thought about that. That approach might create a few problems:

1. Someone does a native ERC20 transfer to the smart contract, such that `_getPrincipalToken().balanceOf(address(this))` is > than `principalRequested`

2. In TermsContract, we have a `require` that checks that `principalDisbursed` is not more than `principalRequested`

I think I can resolve it by removing `totalCrowdfunded`, and instead just passing in the smaller of `balanceOf(address(this))` and `principalRequested` into the `startRepaymentCycle` function

Reply:
https://github.com/enabledao/enable-contracts/pull/51/commits/f1aba44064c75752a077121877d601b20bf37632

Reply:
If we are to use a ceiling, it means we have to implement a cap for the withdrawal function as well, which makes withdrawing in bits more annoying, and would be better we have a simple withdraw function that takes the lesser of `principalRequested` and `balanceOf(address(this))` and transfers the said amount, after which it locks withdrawal

Reply:
Ah what I did was:

* No ceiling on withdrawals (uses `balanceOf(address(this))`)
* Terms Contract will only use a maximum of `principalRequested` to calculate interest and repayments

See https://github.com/enabledao/enable-contracts/commit/f1aba44064c75752a077121877d601b20bf37632